Layer improvement for container and develop container

FedericoMulas opened the pull request:
Started as a fix for the container, ended up as introducing a develop container.

what is changed:
* Added additional unused files to [.dockerignore](https://github.com/WPO-Foundation/wptagent/compare/master...FedericoMulas:wptagent:docker-debug#diff-2f754321d62f08ba8392b9b168b83e24ea2852bb5d815d63e767f6c3d23c6ac5) 
* Few fix to the container and improved layer stacking. With the change code changes will not trigger the apt installation again but will only copy over the new files.
* Added Docker build example  with parameters to [docs/docker.md](https://github.com/WPO-Foundation/wptagent/pull/636/files/8cebfb866805e0fecd6d2153ac0868080a782018#diff-ae5611658dcd443d7a30575bebcdf15a71b7a0f5abebb18b62d00aa0373cae74)
* Added debug state in the Dockerfile definition. This allow to use the container for development/debugging reusing the main container. It leverage the new [wptagent_debug.py](https://github.com/WPO-Foundation/wptagent/compare/master...FedericoMulas:wptagent:docker-debug#diff-5382c96255e6d54e68c36945fc538a91062149626cd88acc93f1af9ef985f931)
* Partial changes from https://github.com/WPO-Foundation/wptagent/pull/625
* Added possibility to pass parameters to the container instead of using environment variables in [docs/docker.md](https://github.com/WPO-Foundation/wptagent/compare/master...FedericoMulas:wptagent:docker-debug#diff-ae5611658dcd443d7a30575bebcdf15a71b7a0f5abebb18b62d00aa0373cae74). Ideally we don't really need all the additional environment variable with this change, but I did not removed them for backward compatiblity.
* Added examples for the new docker stage into the [docs/docker.md](https://github.com/WPO-Foundation/wptagent/pull/636/files/8cebfb866805e0fecd6d2153ac0868080a782018#diff-ae5611658dcd443d7a30575bebcdf15a71b7a0f5abebb18b62d00aa0373cae74)

what is new:
* Added [.gitattributes](https://github.com/WPO-Foundation/wptagent/pull/636/files#diff-618cd5b83d62060ba3d027e314a21ceaf75d36067ff820db126642944145393e) to avoid having to fix line endings on sh files on windows.
* Added [wptagent_debug.py](https://github.com/WPO-Foundation/wptagent/compare/master...FedericoMulas:wptagent:docker-debug#diff-5382c96255e6d54e68c36945fc538a91062149626cd88acc93f1af9ef985f931) for starting the agent in debug mode (useful for the debug container).Foundation/wptagent/compare/master...FedericoMulas:wptagent:docker-debug#diff-ae5611658dcd443d7a30575bebcdf15a71b7a0f5abebb18b62d00aa0373cae74) to explain better the changes.

mjkozicki commented on docs/docker.md:
```diff
@@ -1,45 +1,78 @@
+# Docker Linux Headless Agent
 
-To run the agent, simply specify a few environment variables with docker:
+The `Dokerfile` has multi-stage definition:
```
```suggestion
The `Dockerfile` has multi-stage definition:
```

mjkozicki commented on docs/docker.md:
```diff
@@ -1,45 +1,78 @@
-To run the agent, simply specify a few environment variables with docker:
+The `Dokerfile` has multi-stage definition:
+* **production**: Default stage, produce a image without debug features;
+* **debug**: When running the produced image the wptagent script will wait for a debug to.attach
```
```suggestion
* **debug**: When running the produced image the wptagent script will wait for a debugger to.attach
```

mjkozicki commented on docs/docker.md:
```diff
@@ -1,45 +1,78 @@
-* `SHAPER`: will be passed as `--shaper` (optional)
-* `EXTRA_ARGS`: extra command-line options that will be passed through verbatim (optional)
+Arguments can be passed at build time:
+* **TIMEZONE**: to set the timezone inside the container
```
Is `TIMEZONE` the only argument available at build time? Can you add sample values?

Maybe have the environment variables up here or before the arguments.

FedericoMulas replied:
there was also `DEBIAN_FRONTEND` but I removed it since I don't see any value in allowing interaction during installation at container build time.

I'll add an example.

mjkozicki commented on docs/docker.md:
````diff
@@ -1,45 +1,78 @@
+To build the debug container
+```bash
+docker build --target debug --tag wptagent-debug .
+```
````
```suggestion
```

```

FedericoMulas replied:
this is closing the bash above, why are you suggesting to remove it?

mjkozicki commented on wptagent_debug.py:
```diff
@@ -0,0 +1,16 @@
+    debugpy.wait_for_client()
+    main()
+    # Force a hard exit so unclean threads can't hang the agent
+    os._exit(0)
```
```suggestion
    os._exit(0)

```

FedericoMulas commented:
what is changed:
* Added Docker build example  with parameters to [docs/docker.md](https://github.com/WPO-Foundation/wptagent/pull/636/files/8cebfb866805e0fecd6d2153ac0868080a782018#diff-ae5611658dcd443d7a30575bebcdf15a71b7a0f5abebb18b62d00aa0373cae74)
what is new:
* Added [.gitattributes](https://github.com/WPO-Foundation/wptagent/pull/636/files#diff-618cd5b83d62060ba3d027e314a21ceaf75d36067ff820db126642944145393e) to avoid having to fix line endings on sh files on windows.
